src/datepicker/datepicker.component.spec.ts

```diff
@@ -0,0 +1,115 @@
+import { TestBed } from "@angular/core/testing";
+import { By, EVENT_MANAGER_PLUGINS } from "@angular/platform-browser";
+import { Component, Output, EventEmitter } from "@angular/core";
+import { DatePicker } from "./datepicker.component";
+import { DatePickerInput } from "../datepicker-input/datepicker-input.component";
+import { Calendar16Module } from "@carbon/icons-angular/lib/calendar/16";
+import { FormsModule } from "@angular/forms";
+
+@Component({
+	template: `
+	<ibm-date-picker
+		label="Date picker label"
+		placeholder="mm/dd/yyyy"
+		theme="light"
+		[value]="value"
+		[disabled]="disabled"
+		[invalid]="invalid"
+		invalidText="invalid text"
+		dateFormat="m/d/Y"
+		(valueChange)="valueChange($event)">
+	</ibm-date-picker>
+	`
+})
+class DatePickerTest {
+	value = ["10/19/2019"];
+	disabled = false;
+	invalid = false;
+	valueChange(event) {}
+}
+
+describe("DatePicker", () => {
+	let fixture, element, wrapper;
+	beforeEach(() => {
+		TestBed.configureTestingModule({
+			declarations: [
+				DatePicker,
+				DatePickerTest,
+				DatePickerInput
+			],
+			imports: [
+				Calendar16Module,
+				FormsModule
+			]
+		});
+	});
+
+	it("should work", () => {
+		fixture = TestBed.createComponent(DatePicker);
+		expect(fixture.componentInstance instanceof DatePicker).toBe(true);
+	});
+
+	it("should set label to Date picker label", () => {
+		fixture = TestBed.createComponent(DatePickerTest);
+		element = fixture.debugElement.query(By.css("ibm-date-picker"));
+		fixture.detectChanges();
+		expect(element.nativeElement.textContent).toEqual("Date picker label");
+	});
+
+	it("should set input value to 10/19/2019", () => {
+		fixture = TestBed.createComponent(DatePickerTest);
+		fixture.detectChanges();
+		element = fixture.debugElement.query(By.css("input"));
+		expect(element.nativeElement.value).toEqual("10/19/2019");
+	});
+
+	it("should set placeholder to mm/dd/yyyy", () => {
+		fixture = TestBed.createComponent(DatePickerTest);
+		fixture.detectChanges();
+		element = fixture.debugElement.query(By.css("input"));
+		expect(element.nativeElement.placeholder).toEqual("mm/dd/yyyy");
+	});
+
+	it("should disable input", () => {
+		fixture = TestBed.createComponent(DatePickerTest);
+		fixture.componentInstance.disabled = true;
+		fixture.detectChanges();
+		element = fixture.debugElement.query(By.css("input"));
+		expect(element.nativeElement.disabled).toBe(true);
+	});
+
+	it("should set date format to m/d/Y", () => {
+		fixture = TestBed.createComponent(DatePickerTest);
+		element = fixture.debugElement.query(By.css("ibm-date-picker"));
+		fixture.detectChanges();
+		expect(element.componentInstance.dateFormat).toEqual("m/d/Y");
+	});
+
+	it("should set theme to light", () => {
```

Comment:
There used to be a test for the `dark` theme ... is that something we could add?

Reply:
A class is introduced when the theme is set to light:
![Screen Shot 2019-09-27 at 2 33 26 PM](https://user-images.githubusercontent.com/45505172/65793469-e4d11480-e133-11e9-9650-e78b7c309b47.png)

But remains default when set to dark:
![Screen Shot 2019-09-27 at 2 33 40 PM](https://user-images.githubusercontent.com/45505172/65793495-f5818a80-e133-11e9-8e6d-eea174a32b26.png)

I thought it may be more interesting to test for the existence of the light class.

Reply:
Ah perfect, that makes sense